core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/sql/asterixdb/AsterixDBSourceOpExec.scala

```diff
@@ -201,15 +205,49 @@ class AsterixDBSourceOpExec private[asterixdb] (
     * @throws IllegalArgumentException if attribute does not support string based search
     */
   @throws[IllegalArgumentException]
-  def addKeywordSearch(queryBuilder: StringBuilder): Unit = {
+  def addFilterConditions(queryBuilder: StringBuilder): Unit = {
+    if (search.getOrElse(false) && searchByColumn.isDefined && keywords.isDefined) {
+      val columnType = schema.getAttribute(searchByColumn.get).getType
 
-    val columnType = schema.getAttribute(searchByColumn.get).getType
+      if (columnType == AttributeType.STRING) {
 
-    if (columnType == AttributeType.STRING) {
+        queryBuilder ++= " AND ftcontains(" + searchByColumn.get + ", " + keywords.get + ") "
+      } else
+        throw new IllegalArgumentException("Can't do keyword search on type " + columnType.toString)
+    }
+
+    if (searchByColumnForRegex.isDefined && regex.isDefined) {
+      val regexColumnType = schema.getAttribute(searchByColumnForRegex.get).getType
+      if (regexColumnType == AttributeType.STRING) {
+        queryBuilder ++= "AND regexp_contains(" + searchByColumnForRegex.get + ", \"" + regex.get + "\") "
```

Comment:
there could be a potential bug here, make sure to leave a space before statement snippets, for example, the `AND` here should be ` AND`.

Reply:
done